app/src/main/java/org/mozilla/tv/firefox/webrender/WebRenderHintViewModel.kt

```diff
@@ -31,57 +31,47 @@ class WebRenderHintViewModel(
 ) : ViewModel(), HintViewModel {
 
     override val isDisplayed by lazy {
-        Observable.merge(webRenderOpenedEvents, cursorEvents, loadCompleteEvents)
+        val showEvents = Observable.merge(
+                webRenderOpened,
+                loadComplete,
+                scrollUpOrDownToEndOfDom
+        ).map { true }
+
+        val hideEvents = cursorMovedUpOrDown
+                .map { false }
+
+        Observable.merge(
+                showEvents,
+                hideEvents
+        )
                 .distinctUntilChanged()
                 .replay(1)
                 .autoConnect(0)
     }
     override val hints: Observable<List<HintContent>> = Observable.just(listOf(openMenuHint))
 
-    /**
-     * Emits true when the hint bar should be shown, or false when it should be hidden
-     */
-    private val webRenderOpenedEvents by lazy {
-        screenController.currentActiveScreen
+    private val webRenderOpened = screenController.currentActiveScreen
```

Comment:
Why not lazy anymore?

Reply:
I'm using lazy when the compiler gets unhappy that I'm declaring variables out of order (I want to organize top to bottom on importance, it wants them all declared before usage).  It's not a great solution, but it's the best I've come up with so far.

With the change, this one didn't need that anymore.